### DISCUSSION TITLE
Virtual text appended multiple times

### DISCUSSION THREAD

**Discussion by alxndr** [vuki656/package-info.nvim#97] [General] [answered]
Hi y'all, just noticed an odd bug when using a [`vim-floaterm`](https://github.com/voldikss/vim-floaterm) shortcut to open and then hide a pop-up window: the virtual text for out-of-date packages gets re-appended to the existing virtual text once the pop-up window is closed.

Video:

https://user-images.githubusercontent.com/174307/145490046-7c88ab02-e013-4f2f-8e3c-84801e927890.mov

```
:version
NVIM v0.6.0
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by brew@BigSur

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/Cellar/neovim/0.6.0/share/nvim"
```

**Comment by vuki656** [comment]
Just saw that myself a few days ago. I'll try to fix it over the weekend. 

Thanks for the report. Its tracked in #96

**Comment by vuki656** [comment, marked as answer]
Should be fixed in #99. Let me know if it still occurs.

**Reply by alxndr** [reply]
Looks fixed to me!

However I also notice something perhaps related: in a narrow split, after I use `Ctrl-u` or `Ctrl-d` to move around in the `package.json`, I sometimes see the right-most column in the split showing a handful of characters from the previous cursor position. Kinda hard to describe so here's a video, note the "leftover" blue colons and periods in the right-most edge of the left split:

https://user-images.githubusercontent.com/174307/145904205-92671ded-e4ab-4de3-ae0b-8a8954f927fc.mov

**Reply by vuki656** [reply]
That's weird. 

Could you try to reproduce this with a minimal rc file like [this](https://github.com/hrsh7th/nvim-cmp/blob/main/utils/vimrc.vim) one? Just remove `cmp` related stuff and place this package instead.

**Reply by alxndr** [reply]
https://github.com/alxndr/dotfiles/blob/26391db30/vimrc-minimal.vim

`$ nvim -u vimrc-minimal.vim package.json`

Then I run `:lua require('package-info').show({force=true})` and it errors, this is what is in `:messages`

```
...plug/package-info.nvim/lua/package-info/modules/core.lua:220: Expected lua number
stack traceback:
^I[C]: in function 'nvim_buf_set_extmark'
^I...plug/package-info.nvim/lua/package-info/modules/core.lua:220: in function '__set_virtual_text'
^I...plug/package-info.nvim/lua/package-info/modules/core.lua:236: in function '__display_virtual_text'
^I...plug/package-info.nvim/lua/package-info/modules/core.lua:275: in function 'on_success'
^I...ed/vim-plug/package-info.nvim/lua/package-info/utils.lua:88: in function <...ed/vim-plug/package-info.nvim/lua/package-info/utils.lua:74>
```

**Reply by vuki656** [reply]
Try running `require('package-info').setup()` before running the `show` command
